sktime/utils/seasonality.py

```diff
@@ -111,24 +111,22 @@ def _pivot_sp(df, sp, anchor=None, freq=None, anchor_side="start"):
     if freq is None and hasattr(anchor.index, "freq"):
         freq = anchor.index.freq
 
-    if isinstance(df.index, pd.DatetimeIndex):
-        df = df.copy()
-        df.index = df.index.to_period(freq=freq)
-        was_datetime = True
-    else:
-        was_datetime = False
+    df, was_datetime = _make_period_index_df(df, freq)
 
     if isinstance(anchor.index, pd.DatetimeIndex):
         aix = anchor.index.to_period(freq=freq)
     else:
         aix = anchor.index
     aix_int = aix.astype("int64")
 
+    if pd.api.types.is_integer_dtype(anchor.index) or len(anchor) <= 1:
+        period_len = 1
+        period_len_int = 1
+    else:
+        period_len = anchor.index[1] - anchor.index[0]
+        period_len_int = aix_int[1] - aix_int[0]
+
     if not isinstance(df.index, pd.PeriodIndex):
-        if pd.api.types.is_integer_dtype(anchor.index) and len(anchor) <= 1:
```

Comment:
can you explain why you deleted these 4 lines?

Reply:
I just moved them before the first if condition, to avoid code duplications I introduced. `period_len_int` is required to get the correct indices when calculating the pivot_table. Since the index that results from `aix_int = aix.astype("int64")` is not consecutive if we have a non one frequency. Thus, the sp needs to be multiplied with the correct `period_len_int` 

When explaining that, I think I made the assumption that if sp=1 but the frequency of the time series is 2 weeks than the seasonality should be 2 weeks too.

Reply:
Probably, this is not working for monthly time series, since then there would not be an unambiguously period_len_int.... 
See: 
```python
monthly_int_index = pd.date_range("2020.01.01", freq="M", periods=5).astype("int")
monthly_int_index[1:] - monthly_int_index[:-1]
# output:  Index([1580428800000000000, 1582934400000000000, 1585612800000000000, 1588204800000000000, 1590883200000000000], dtype='int64')
```

Reply:
😱